dungeonGen.js

```diff
@@ -0,0 +1,284 @@
+$(function() {
+'use strict';
+
+	var sizeOpts, 
+		sizeChoice,
+		tSize,
+		points, 
+		posit, 
+		tile = $('canvas')[0].getContext('2d'), 
+		tileOut = $('canvas')[0].getContext('2d'),
+		width = $('canvas').width(), 
+		x = width/2,
+		height = $('canvas').height(), 
+		y = height/2,
+		coords = {"xPos":x, "yPos":y}; //default coordinates
+
+	//the initial black background
+	tile.fillStyle = "rgb(0,0,0)";
+	tile.fillRect(0, 0, width, height);
+
+	//disable the editing buttons upon app launch
+	$('#addButton').prop('disabled', true);
+	$('#undoButton').prop('disabled', true);
+
+	//Define a tile object
+	function Tile() {
+		this.inStack = [];
+		this.tileSize = tSize;
+
+		//find the index of the input coordinates in the tile array inStack
+		this.whereInStack = function(posit) {
+			for(var i = 0; i < this.inStack.length; i ++){
+				if(this.inStack[i][0] == posit[0] && this.inStack[i][1] == posit[1])
+					return i;
+			}
+			return -1;
+		}
+	}
+
+	//dungeon floor tile
+	function floorTile(color) {
+		this.color = color;
+		this.tileType = 'floor';
+	}
+
+	//setting the pointer to the dungeon floor tile's prototype to be the tile object
+	floorTile.prototype = new Tile();
```

Comment:
I intend on having more tile objects later, but it might actually not happen...So for now I will just throw those properties into Tile.

I also wanted to play with inheritance in JS haha